x-pack/plugins/infra/public/pages/logs/stream/page_logs_content.tsx

```diff
@@ -291,12 +295,13 @@ type InitializedLogStreamPageState = MatchedStateFromActor<
 
 export const StreamPageLogsContentForState = React.memo<{
   logStreamPageState: InitializedLogStreamPageState;
-}>(({ logStreamPageState }) => {
+  logStreamPageSend: LogStreamPageSend;
+}>(({ logStreamPageState, logStreamPageSend }) => {
   const {
     context: { parsedQuery },
   } = logStreamPageState;
 
-  return <StreamPageLogsContent filterQuery={parsedQuery} />;
+  return <StreamPageLogsContent filterQuery={parsedQuery} logStreamPageSend={logStreamPageSend} />;
```

Comment:
By extracting the `parsedQuery` we remove knowledge of the `logStreamPageState`, but the `send` gets passed on unchanged.

Not necessarily asking for a change here if you think this would become too cluttered. :innocent: I just wonder if this would be a good place to turn the generic `logStreamPageSend` into multiple specific callbacks so that this layer prevents all knowledge about the existence of state machines from the content component?

Reply:
> turn the generic logStreamPageSend into multiple specific callbacks so that this layer prevents all knowledge about the existence of state machines from the content component?

That sounds like a nice improvement. Clutter probably isn't too much of an issue, potentially we could end up with a lot of callbacks but to be honest I think given the low complexity of them seeing a lot of them isn't really an issue. If they do grow too large we could always do something with composition (like maybe one of these handling each machine, but still at this `page_logs_content` of the hierarchy).

Reply:
prior art for this would be the `mapToDispatch` in redux's `connect()` helper, which could be seen as serving the same purpose as this connection component